Automattic/jetpack
# Jetpack Sync: Ensure duplicate Sync modules are not loaded

This PR ensures that no matter the consumer Sync configuration or usage of the `jetpack_sync_modules` filter, no duplicate Sync modules will be loaded.

## Proposed changes:

* `Automattic\Jetpack\Sync\Modules`: Update `initialize_modules` to ensure the list of Sync module classnames returned after applying the `jetpack_sync_modules` filter is unique.
* Update Sync's README to make it clear that only the default Sync modules are configurable by the consumers.

### Other information:


## Jetpack product discussion


p9dueE-8kC-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






Pre-requisites: A JN site with WooCommerce and WooCommerce Payments plugins active (latest stable versions). We'll also need the Jetpack Beta plugin to switch between Bleeding Edge and this branch.

Attention: Make sure to setup your store by running the WooCommerce setup wizard before testing.

With the Jetpack plugin on Bleeding Edge:
* Run the following in `wp shell`: `return Automattic\Jetpack\Sync\Modules::get_modules()`
* Check the list of Sync modules and confirm that `WooCommerce` and `WooCommerce_HPOS_Orders` are listed twice

With the current branch:
* Repeat the above and confirm that `WooCommerce` and `WooCommerce_HPOS_Orders` are listed once (you'll need to exit `wp shell` first)
